main.go

```diff
@@ -92,22 +105,65 @@ func printVersion() {
 }
 
 var (
-	errNoPackageFound = errors.New("no package found")
+	errNoPackageFound        = errors.New("no package found")
+	errUnsupportMultiplePkgs = errors.New("multiple packages are not supported, please specify the package ID or directory path that only contains one package")
 )
 
 func generateFromRequest(request GenerationRequest) (string, error) {
 	cfg := packages.Config{
 		Mode: packages.NeedName | packages.NeedTypes,
 	}
-	pkgs, err := packages.Load(&cfg, request.Pkg)
+	if request.ModelPkg == "" || request.ModelPkg == request.Pkg {
```

Comment:
Remove this `if`. Handle case `modelPkg == repoPkg` the same way as `modelPkg != repoPkg`